cmd/k-license/k-license.go

```diff
@@ -0,0 +1,259 @@
+	".java":  true,
+	".cpp":   true,
+	".sh":    true,
+}
+
+var buildFileExts = map[string]bool{
+	"Makefile":   true,
+	"Dockerfile": true,
+}
+
+type Options struct {
+	templatesDir string
+	excludeDirs  []string
+	path         string
+	confirm      bool
+}
+type templateFileType struct {
+	fileExtension    string // store file extension strings like ".sh", "Makefile", etc.
+	templateFileName string // store template file names like "boilerplate.sh.txt", etc
+}
+
+var templateFileTypes = []templateFileType{
+	{".sh", "boilerplate.sh.txt"},
+	{"Makefile", "boilerplate.Makefile.txt"},
+	{"Dockerfile", "boilerplate.Dockerfile.txt"},
+	{".py", "boilerplate.py.txt"},
+	{".go", "boilerplate.go.txt"},
+}
+
+var opts = &Options{}
+
+func main() {
+	rootCmd := &cobra.Command{
+		Use:   "k-license",
+		Short: "Tool for Adding license Headers",
+	}
+	addCmd := &cobra.Command{
+		Use:   "add",
+		Short: "Add Headers to files",
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			return opts.Run()
+		},
+	}
+
+	addCmd.Flags().StringVar(&opts.templatesDir, "templates", "../../hack/boilerplate", "directory containing license templates")
+	addCmd.Flags().StringSliceVarP(&opts.excludeDirs, "exclude", "e", excludeDirsLocations, "comma-separated list of directories to exclude")
+	addCmd.Flags().StringVar(&opts.path, "path", ".", "Defaults to Current directory")
+	addCmd.Flags().BoolVar(&opts.confirm, "confirm", false, "")
+	rootCmd.AddCommand(addCmd)
+	if err := rootCmd.Execute(); err != nil {
+		os.Exit(1)
+	}
+
+}
+func (opts *Options) Run() error {
+	//excludeList := strings.Split(opts.excludeDirs, ",")
+	files := 0
+	fileList := make([]string, 0)
+	err := filepath.WalkDir(opts.path, func(path string, info fs.DirEntry, err error) error {
+		if info.IsDir() && containsExcluded(opts.excludeDirs, info.Name()) {
+			return filepath.SkipDir
+		}
+		if !info.IsDir() && (isCodeFile(path) || isBuildFile(path)) && !isGenerateFile(path) {
+			hasLic, err := hasLicense(path)
+			if !hasLic {
+				currentYear := strconv.Itoa(time.Now().Year())
+				if opts.confirm {
+					err := addLicense(path, opts.templatesDir, currentYear)
+					if err != nil {
+						return err
+					}
+					fmt.Printf("Modified %s file\n", path)
+				}
+				fileList = append(fileList, path)
+				files++
+			}
+			return err
+
+		}
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+	if opts.confirm {
+		fmt.Printf("Modified %v files\n", files)
+	} else {
+		fmt.Printf("DRY RUN: No file changes will be made! To make file modifications, rerun the command with  \"--confirm\" flag\n")
+		if files == 0 {
+			fmt.Printf("All files have appropriate License Headers. No changes required.\n")
+		}
+		if files > 0 {
+			fmt.Printf("%v files will be modified to add License Headers\n", files)
+			fmt.Printf("Listing files to be modified:\n")
+			for _, file := range fileList {
+				fmt.Printf("%s\n", file)
+			}
+		}
+
+	}
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
```

Comment:
Let's return the error here, `cobra` will exit with a non 0 exit code if an error is returned to `RunE`.